### PR TITLE
Add OpenMathInstruct-1

### PR DESCRIPTION
**Pull request by mmhamdy**
Hi,
This PR adds OpenMathInstruct-1 (with the downloader, preparer, and mapper). All tests passed successfully.

Note: I have added a new license to the license classes based on the information provided [here](https://huggingface.co/datasets/nvidia/OpenMathInstruct-1/blob/main/LICENSE)

### CODE REVIEW

File: constants/license_classes.json [hunk 1]
```diff
@@ -174,6 +174,11 @@
         "0",
         "0"
     ],
+    "NVIDIA License": [
```
Comment (shayne-longpre) [inline constants/license_classes.json:177]: I think we should add this to custom licenses, as it is a NVIDIA license, but not the only type of NVIDIA license, if I understand correctly?

File: data_summaries/OpenMathInstruct-1.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            "Mathematical Equation Generation",
+            "Probabilistic Reasoning",
+            "Question Answering"    
+        ],
+        "Text Sources": [
+            "artofproblemsolving.com/community/c3158_usa_contests",
+            "american invitational mathematics examination (AIME)",
+            "american mathematics competitions (AMC)",
+            "mathematics competitions",
+            "math"
```
Comment (shayne-longpre) [inline data_summaries/OpenMathInstruct-1.json:30]: Is this a website/book/text source? Or crowdsourced math questions? We probably don't need to add just "math", but instead the source or "crowdsourced"

Comment (shayne-longpre) [reply data_summaries/OpenMathInstruct-1.json:30]: Oh I see it's a dataset. Fine to just have the derived from "math" dataset in that part only

**Review by shayne-longpre — APPROVED** [review]
Looks good aside from a column minor comments to fix

File: data_summaries/OpenMathInstruct-1.json (hunk 2 above)
Comment (mmhamdy) [reply data_summaries/OpenMathInstruct-1.json:30]: Yes, the math in the text sources refers to the "MATH" dataset, and AIME and AMC are where MATH got the problems. Should I keep math only as a text source?

File: constants/license_classes.json (hunk 1 above)
Comment (mmhamdy) [reply constants/license_classes.json:177]: Probably, I couldn't find a link for the definition.

**Pull request merged**